src/branding.ts

```diff
@@ -26,60 +46,149 @@ import {
  * when you know you need it. If doing an equality check, it's almost always
  * better to use {@linkcode StrictEqualUsingTSInternalIdenticalToOperator}.
  */
-export type DeepBrand<T> =
+export type DeepBrand<T, Options extends DeepBrandOptions> =
   IsNever<T> extends true
     ? {type: 'never'}
     : IsAny<T> extends true
       ? {type: 'any'}
       : IsUnknown<T> extends true
         ? {type: 'unknown'}
-        : T extends string | number | boolean | symbol | bigint | null | undefined | void
-          ? {
-              type: 'primitive'
-              value: T
-            }
-          : T extends new (...args: any[]) => any
+        : Not<IsNever<NominalType<T, Options>>> extends true
+          ? {type: NominalType<T, Options>}
```

Comment:
I'm sure there's a better way to do this without having to explicitly check for never/user `NominalType` twice